### PR TITLE
Detector is now OAuth2 client for Butler

### PR DESCRIPTION
**Pull request by hedgehog1833**
The detector now has two oauth2 client for the butler, user and admin. It gets its tokens from the new auth service. This PR also fixed the bug that our client credentials tokens (for the spotify search) were currently not persisted and refetched on every search.

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[OAuth2 Metal Detector - Butler](https://trello.com/c/HFHClh95/408-oauth2-metal-detector-butler)

File: docker-compose.yml [hunk 1]
```diff
@@ -57,6 +56,42 @@ services:
...
+      POSTGRES_DB: metal-detector-auth
+      PGDATA: /var/lib/postgresql/data/db-files/
+    networks:
+      - auth-network
+    volumes:
+      - type: volume
+        source: auth-db-volume
+        target: /var/lib/postgresql/data
+    ports:
+      - 5434:5432
```
Comment (DanielW1987) [inline docker-compose.yml:73]:
```suggestion
      - "5434:5432"
```

File: docker-compose.yml [hunk 2]
```diff
@@ -57,6 +56,42 @@ services:
...
+      PRIVATE_KEY:
+      PUBLIC_KEY:
+      METAL_DETECTOR_USER_CLIENT_ID: metal-detector-user
+      METAL_DETECTOR_USER_CLIENT_SECRET: userSecret
+      METAL_DETECTOR_ADMIN_CLIENT_ID: metal-detector-admin
+      METAL_DETECTOR_ADMIN_CLIENT_SECRET: adminSecret
+    networks:
+      - auth-network
+    ports:
+      - 9000:8080
```
Comment (DanielW1987) [inline docker-compose.yml:92]:
```suggestion
      - "9000:8080"
```

File: webapp/src/main/resources/application-prod.yml [hunk 3]
```diff
@@ -29,6 +29,9 @@ spring:
             redirect-uri: "https://{baseHost}{basePort}{basePath}/login/oauth2/code/{registrationId}"
           spotify-user:
             redirect-uri: "https://{baseHost}{basePort}{basePath}/settings/spotify-synchronization"
+        provider:
+          metal-detector-auth:
+            issuer-uri: http://auth-app:8080
```
Comment (DanielW1987) [inline webapp/src/main/resources/application-prod.yml:34]: please use public url here

**Review by DanielW1987 — APPROVED** [review]

**Pull request merged**